src/RadixSortLSD.chpl

```diff
@@ -159,52 +163,55 @@ module RadixSortLSD
         } // for rshift
     }//proc radixSortLSDCore
 
-    proc radixSortLSD(a:[?aD] ?t, checkSorted: bool = true): [aD] (t, int) {
+    proc radixSortLSD(a:[?aD] ?t, checkSorted: bool = true, numTasks: int = here.maxTaskPar, bitsPerDigit: int = RSLSD_bitsPerDigit): [aD] (t, int) {
         var kr: [aD] (t,int) = [(key,rank) in zip(a,aD)] (key,rank);
         if (checkSorted && isSorted(a)) {
             return kr;
         }
         var (nBits, negs) = getBitWidth(a);
-        radixSortLSDCore(kr, nBits, negs, new KeysRanksComparator());
+        var plan = new RadixSortLSDPlan(numTasks, bitsPerDigit);
+        radixSortLSDCore(kr, nBits, negs, new KeysRanksComparator(), plan);
         return kr;
     }
 
     /* Radix Sort Least Significant Digit
        radix sort a block distributed array
        returning a permutation vector as a block distributed array */
-    proc radixSortLSD_ranks(a:[?aD] ?t, checkSorted: bool = true): [aD] int {
+    proc radixSortLSD_ranks(a:[?aD] ?t, checkSorted: bool = true, numTasks: int = here.maxTaskPar, bitsPerDigit: int = RSLSD_bitsPerDigit): [aD] int {
         if (checkSorted && isSorted(a)) {
             var ranks: [aD] int = [i in aD] i;
             return ranks;
         }
 
         var kr: [aD] (t,int) = [(key,rank) in zip(a,aD)] (key,rank);
         var (nBits, negs) = getBitWidth(a);
-        radixSortLSDCore(kr, nBits, negs, new KeysRanksComparator());
+        var plan = new RadixSortLSDPlan(numTasks, bitsPerDigit);
+        radixSortLSDCore(kr, nBits, negs, new KeysRanksComparator(), plan);
         var ranks: [aD] int = [(_, rank) in kr] rank;
         return ranks;
     }
 
     /* Radix Sort Least Significant Digit
        radix sort a block distributed array
        returning sorted keys as a block distributed array */
-    proc radixSortLSD_keys(a: [?aD] ?t, checkSorted: bool = true): [aD] t {
+    proc radixSortLSD_keys(a: [?aD] ?t, checkSorted: bool = true, numTasks: int = here.maxTaskPar, bitsPerDigit: int = RSLSD_bitsPerDigit): [aD] t {
         var copy = a;
         if (checkSorted && isSorted(a)) {
             return copy;
         }
         var (nBits, negs) = getBitWidth(a);
-        radixSortLSDCore(copy, nBits, negs, new KeysComparator());
+        var plan = new RadixSortLSDPlan(numTasks, bitsPerDigit);
+        radixSortLSDCore(copy, nBits, negs, new KeysComparator(), plan);
         return copy;
     }
 
-    proc radixSortLSD_memEst(size: int, itemsize: int) {
+    proc radixSortLSD_memEst(size: int, itemsize: int, numTasks: int = here.maxTaskPar, numBuckets: int = 1 << RSLSD_bitsPerDigit) {
```

Comment:
same comment as before.

Reply:
I'm not as sure about doing this in the `memEst` functions. Because if we move towards having the `RadixSortLSDPlan` contain the arrays, we would want to call the memory estimates before we actually created the `plan`, right? I'll make this change for now, since we're not currently having the plan object store the created arrays and it's easy enough to switch back